api/server/router/container/container_routes.go

```diff
@@ -940,9 +940,9 @@ func (s *containerRouter) postContainersAttach(ctx context.Context, w http.Respo
 			if multiplexed && versions.GreaterThanOrEqualTo(httputils.VersionFromContext(ctx), "1.42") {
 				contentType = types.MediaTypeMultiplexedStream
 			}
-			fmt.Fprintf(conn, "HTTP/1.1 101 UPGRADED\r\nContent-Type: "+contentType+"\r\nConnection: Upgrade\r\nUpgrade: tcp\r\n\r\n")
+			_, _ = fmt.Fprint(conn, "HTTP/1.1 101 UPGRADED\r\nContent-Type: "+contentType+"\r\nConnection: Upgrade\r\nUpgrade: tcp\r\n\r\n")
 		} else {
-			fmt.Fprintf(conn, "HTTP/1.1 200 OK\r\nContent-Type: application/vnd.docker.raw-stream\r\n\r\n")
+			_, _ = fmt.Fprint(conn, "HTTP/1.1 200 OK\r\nContent-Type: application/vnd.docker.raw-stream\r\n\r\n")
```

Comment:
Should we really be ignoring errors encountered while writing to the socket? If the connection has e.g. been reset by the remote end, any attempt to continue handling the request is just wasting cycles.

And on the topic of ignoring write errors, see line 977.

Reply:
I'll have a look at this in a follow-up; I had a quick peek at how this code is used, and saw some other parts that I want to have a look at.